lib/chef/provider/aws_ebs_volume.rb

```diff
@@ -204,4 +204,8 @@ def delete(volume)
       volume
     end
   end
+
+  def availability_zone
+    "#{region}#{new_resource.availability_zone}"
```

Comment:
Now that 1.0 is out, we need to support backwards compatability.  Can you add in a regex that checks if `new_resource.availability_zone` begins with `region`?  If so, it should output a WARN message about the deprecated logic and then only return the provided availability_zone.  Otherwise it should do the logic here.